DataFormats/simulation/src/Stack.cxx

```diff
@@ -208,7 +209,9 @@ TParticle* Stack::PopNextTrack(Int_t& iTrack)
 
   // If end of stack: Return empty pointer
   if (mStack.empty()) {
-    notifyFinishPrimary();
+    if (mParticles.size() > 0) { // make sure something was tracked at all
+      notifyFinishPrimary();
+    }
```

Comment:
This is certainly ok. The only question is how this can happen? How can the simulation be called on empty events ... probably one should also fix side pushing to the stack.

Reply:
@sawenzel Thanks. The Kinematics.root was generated with |eta|<2.5 cut on trackable particles (to save the time). In the event which caused the problem all decay products were outside this range, so they were not imported from the generator.

Reply:
I see. Thanks for the explanation.